tools/bcr_validation.py

```diff
@@ -238,6 +238,31 @@ def get_github_user_id(github_username):
     return None
 
 
+def is_valid_bazel_compatability_for_overlay(bazel_compatibility):
+    """
+    Returns whether the bazel_compatability is valid for an overlay.
+    See: https://bazel.build/rules/lib/globals/module#module
+
+    Args:
+        bazel_compatability: List of bazel compatability strings.
+
+    Returns:
+        Boolean indicating compatability with source overlays.
+    """
+    if not bazel_compatibility:
+        return False
+    for v in bazel_compatibility:
+        m = re.fullmatch(r"^([><-]=?)(\d+\.\d+\.\d+)$", v)
+        if not m or m.group(1) == "-":
+            continue  # Skip - versions
+        version = tuple(int(i) for i in m.group(2).split("."))
+        if m.group(1) == ">":
+            return version > (7, 2, 0)
```

Comment:
Since all conditions are combined with "and" semantics, we should probably have a variable that's set to True here rather than potentially returning False without having seen all entries.

Reply:
This returns False only after all entries are seen.
Since ands can be short-circuited, any entry > 7.2.0 will cause it to return true.

Reply:
Wouldn't this return False directly when it sees `>6.1.0` even if there is a `>7.2.1` that comes after it? But yeah, that won't show up in practice.

Reply:
Fixed to only short circuit True.  Yeah, I wasn't thinking about that.